### DISCUSSION TITLE
Security Trails API wrapper

### DISCUSSION THREAD

**Discussion by hatecomputers** [projectdiscovery/uncover#58] [Ideas]
Would it be possible to have a Security Trails API wrapper just like there is for Shodan, Censys etc? I woud love to toss aside my own wrapper in favor of using uncover instead.

**Comment by ehsandeep** [comment]
@hatecomputers I had no idea they had search engines like shodan and censys, it would be a great addition, do you have API reference for Security Trails **search**?

**Reply by hatecomputers** [reply]
@ehsandeep https://docs.securitytrails.com/docs I guess that would be it? It's slightly different I would say but definitely see advantage of integrating with their API. 

My current usage basically revolves around subdomain extraction which is something like:
```bash
curl -s https://api.securitytrails.com/v1/domain/{domain}/subdomains?children_only=true -H "apiKey: <API_KEY>"
```
but there's definitely more to it as you it can be seen in the docs. Setup should be easy just like any other existing providers since you just basically rely on the API_KEY.

**Reply by ehsandeep** [reply]
@hatecomputers in that case [subfinder](https://github.com/projectdiscovery/subfinder) already does that `subfinder -s securitytrails` where securitytrails is supported for subdomain enum.
